docs/source/tutorial/recurrentnet.rst

```diff
@@ -213,6 +213,111 @@ Since the history of computation is only memorized between variables ``feat`` an
    It is not allowed to mix volatile and non-volatile variables as arguments to same function.
    If you want to create a variable that behaves like a non-volatile variable while can be mixed with volatile ones, use ``'auto'`` flag instead of ``'off'`` flag.
 
+
+Making it with Trainer
+~~~~~~~~~~~~~~~~~~~~~~
+
+The above codes are written with plain Function/Variable APIs.
+When we write a training loop, it is better to use Trainer, since we can then easily add functionalities by extensions.
+
+Before implementing it on Trainer, let's clarify the training settings.
+We here use Penn Tree Bank dataset as a set of sentences.
+Each sentence is represented as a word sequence.
+We concatenate all sentences into one long word sequence, in which each sentence is separated by a special word ``<eos>``.
+This dataset is easily obtained by :func:`chainer.datasets.get_ptb_words`.
+This function returns train, validation, and test dataset, each of which is represented as a long array of integers.
+Each integer represents a word ID.
+
+Our task is to learn a recurrent neural net language model from the long word sequence.
+We use words in different locations to form mini-batches.
+It means we maintain :math:`B` indices pointing to different locations in the sequence, read from these indices at each iteration, and increment all indices after the read.
+Of course, when one index reaches the end of the whole sequence, we turn the index back to 0.
+
+In order to implement this training procedure, we have to customize the following components of Trainer:
+
+- Iterator.
+  Built-in iterators do not support reading from different locations and aggregating them into a mini-batch.
+- Update function.
+  The default update function does not support truncated BPTT.
+
+When we write a dataset iterator dedicated to the dataset, the dataset implementation can be arbitrary; even the interface is not fixed.
+On the other hand, the iterator must support the :class:`~chainer.dataset.Iterator` interface.
+The important methods and attributes to implement are ``batch_size``, ``epoch``, ``epoch_detail``, ``is_new_epoch``, ``iteration``, ``__next__``, and ``serialize``.
+Following is a code from the official example in the ``examples/ptb`` directory.
+
+.. code-block:: python
+
```

Comment:
Currently no, because it is too specific to this task. If some users want to use it outside of this example, we can consider move it to the package.